src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,137 @@
+            current = current.nextLinkedList;
+        }
+        Node<T> next = new Node<>(null, value, current);
+        current.nextLinkedList = next;
+        count++;
+        return true;
     }
 
     @Override
-    public void add(T value, int index) {
-
+    public void add(T value,int index) {
+        if (index < 0 || index > count) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (index == count) {
+            add(value);
+            return;
+        }
+        Node<T> current = ourNode;
+        for (int i = 0; i < index; i++) {
+            current = current.nextLinkedList;
+        }
+        Node<T> prevNode = current.prevLinkedList;
+        Node<T> newNode = new Node<>(current, value, prevNode);
+        current.prevLinkedList = newNode;
+        prevNode.nextLinkedList = newNode;
+        count++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T l: list) {
+            add(l);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = ourNode;
+        for (int i = 0; i < index; i++) {
+            current = current.nextLinkedList;
+        }
+        return current.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = ourNode;
+        for (int i = 0; i < index; i++) {
+            current = current.nextLinkedList;
+        }
+        T result = (T) current.item;
+        current.item = value;
+        return result;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = ourNode;
+        for (int i = 0; i < index; i++) {
+            current = current.nextLinkedList;
+        }
+        T result = current.item;
+        Node<T> before = current.prevLinkedList;
+        Node<T> after = current.nextLinkedList;
+        if (before != null && after != null) {
+            before.nextLinkedList = after;
+            after.prevLinkedList = before;
+        } else if (before == null) {
+            ourNode = after;
+        }
+        count--;
+        return result;
     }
 
     @Override
     public boolean remove(T t) {
+        Node<T> current = ourNode;
+        for (int i = 0; i < count; i++) {
+            T el = current.item;
+            if (t == null ? el == null : el.equals(t)) {
+                remove(i);
+                return true;
+            }
+            current = current.nextLinkedList;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return count;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return count == 0;
+    }
+
+    public class Node<T> {
+        private T item;
+        private Node<T> nextLinkedList;
```

Comment:
```suggestion
        private Node<T> nextNode;
```
it's gonna be better cause we have only one linkedList with nodes inside it